app/lib/brexit_checker/results_audiences.rb

```diff
@@ -1,65 +1,68 @@
 class BrexitChecker::ResultsAudiences
-  class << self
-    def populate_business_groups(audience_actions, selected_criteria)
-      return {} if audience_actions.blank? || selected_criteria.blank?
+  attr_reader :citizen_actions, :business_actions, :criteria
 
-      {
-        actions: audience_actions,
-        criteria: audience_actions.flat_map(&:all_criteria).uniq & selected_criteria,
-      }
-    end
+  def initialize(actions, criteria)
+    @citizen_actions = actions["citizen"]
+    @business_actions = actions["business"]
+    @criteria = criteria
+  end
 
-    def populate_citizen_groups(audience_actions, selected_criteria)
-      return [] if audience_actions.blank? || selected_criteria.blank?
+  def populate_business_groups
+    return {} if business_actions.blank? || criteria.blank?
 
-      audience_actions.each_with_object([]) do |action, grouped_actions|
-        next if action.grouping_criteria.empty?
+    {
+      actions: business_actions,
+      criteria: business_actions.flat_map(&:all_criteria).uniq & criteria,
+    }
+  end
 
-        group_keys =
-          multiple_grouping_criteria?(action) ? selected_group(action, selected_criteria) : action.grouping_criteria
-        group_keys.each do |key|
-          group = all_groups.select { |g| g.key == key }.first
-          next if grouped_actions.select { |actions| actions[:group] == group }.any?
+  def populate_citizen_groups
```

Comment:
This method is still pretty complicated. Any chance we could break it down a little?